### PR TITLE
[synthetics-private-location] add ability to template config name

### PR DESCRIPTION
**Pull request by iorubs**
#### What this PR does / why we need it:

Ability to supply a template for the configSecret.
* Reference secrets that were created with a template without having to maintain multiple values files.
* Easier to integrate with external secret providers.

#### Checklist
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has been updated

### CODE REVIEW

File: charts/synthetics-private-location/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Datadog changelog
 
+### 0.14.2
+
+* Added ability to template the config secret name.
```
Comment (badouralix) [inline charts/synthetics-private-location/CHANGELOG.md:5]:
```suggestion
* Add ability to template the config secret name.
```

**Review by badouralix — CHANGES_REQUESTED** [review]
Hi @iorubs, thanks for your contribution ! Overall lgtm, just added one nitpick

Would to be able to make a similar change for `configConfigMap` a few lines above ?

File: charts/synthetics-private-location/CHANGELOG.md (hunk 1 above)
Comment (iorubs) [reply charts/synthetics-private-location/CHANGELOG.md]: Done.

**Comment by iorubs** [comment]
> Hi @iorubs, thanks for your contribution ! Overall lgtm, just added one nitpick
> 
> Would to be able to make a similar change for `configConfigMap` a few lines above ?

Done.

**Review by badouralix — APPROVED** [review]

**Comment by iorubs** [comment]
@badouralix one of the checks is timing out, do you have any recommendation on what I should do to fix/rerun it?

**Comment by badouralix** [comment]
So the chart installation is failing due to the following error, which sort of makes sense since the `synthetics-private-location` chart does not provide the configmap itself and expects it to be created elsewhere

```text
MountVolume.SetUp failed for volume "worker-config" : configmap "config-map-name" not found
```

With the current test setup, creating the configmap/secret before installing the chart is a bit tricky

@iorubs let's remove the new files under `ci/` for now
We will add them back in another pull request once we can handle the configmap/secret creation

**Comment by iorubs** [comment]
> So the chart installation is failing due to the following error, which sort of makes sense since the `synthetics-private-location` chart does not provide the configmap itself and expects it to be created elsewhere
> 
> ```
> MountVolume.SetUp failed for volume "worker-config" : configmap "config-map-name" not found
> ```
> 
> With the current test setup, creating the configmap/secret before installing the chart is a bit tricky
> 
> @iorubs let's remove the new files under `ci/` for now We will add them back in another pull request once we can handle the configmap/secret creation

That makes sense, thanks.

Done.

**Review by badouralix — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]

**Pull request merged**